app/uk/gov/hmrc/agentsubscription/controllers/RegistrationController.scala

```diff
@@ -44,14 +48,19 @@ class RegistrationController @Inject()(service: RegistrationService, override va
       service.getRegistration(utr, postcode).map(_
         .map(registrationDetails => Ok(toJson(registrationDetails)))
         .getOrElse(NotFound))
+
   }
 
   private def badRequest(code: String) = {
     Future successful BadRequest(Json.obj("code" -> code))
   }
 
-  def getRegistration(utr: Utr, postcode: String): Action[AnyContent] =
-    withAgentAffinityGroup.async(getRegistrationBlock(utr, postcode))
+  def getRegistration(utr: Utr, postcode: String) = forRegistration {
```

Comment:
I find names like `forRegistration`a little bit too abstract, does it reveals what it does without forcing a reader to look deep inside the code?

Reply:
renamed "forRegistration" to "affinityGroupAndCredentials"
renamed "forSubscription" to "affinityGroupAndEnrolments"